### PR TITLE
feat: integrate auth designs with bloc logic

### PR DESCRIPTION
**Pull request by wizlif**

### CODE REVIEW

File: lib/infrastructure/auth/firebase_auth_repository.dart [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    try {
+      final String verificationId = authCredentials.verificationId!;
+      final String smsCode = authCredentials.smsCode!;
+      // Create a PhoneAuthCredential with the code
+      final credential = fb_auth.PhoneAuthProvider.credential(
+          verificationId: verificationId, smsCode: smsCode);
+
+      // Sign the user in (or link) with the credential
+      await firebaseAuth.signInWithCredential(credential);
+      //TODO Check if is new user
```
Comment (Xazin) [inline lib/infrastructure/auth/firebase_auth_repository.dart:91]: Do we need a new commit for this, or can we do it in this PR?

File: lib/main.dart [hunk 2]
```diff
@@ -17,7 +16,7 @@ Future<void> main() async {
   runApp(AppWidget());
 }
 
-class AppWidget extends StatelessWidget {
+/*class AppWidget extends StatelessWidget {
```
Comment (Xazin) [inline lib/main.dart:19]: Remove commented code when you move to a new file.

File: lib/presentation/auth/_dispose/register_phone_number_screen.dart [hunk 3]
```diff
@@ -0,0 +1,88 @@
+/*import 'package:auto_route/auto_route.dart';
```
Comment (Xazin) [inline lib/presentation/auth/_dispose/register_phone_number_screen.dart:1]: Delete file? We have a git history, we can see files in older commits.

File: lib/presentation/auth/_dispose/verify_code_screen.dart [hunk 4]
```diff
@@ -1,8 +1,8 @@
-import 'package:flutter/material.dart';
```
Comment (Xazin) [inline lib/presentation/auth/_dispose/verify_code_screen.dart:1]: Delete file?

File: lib/presentation/auth/auth_screen.dart [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+class AuthPage extends StatefulWidget {
+  const AuthPage({Key? key}) : super(key: key);
+
+  @override
+  _AuthPageState createState() => _AuthPageState();
+}
+
+class _AuthPageState extends State<AuthPage> {
+  // All Pages
```
Comment (Xazin) [inline lib/presentation/auth/auth_screen.dart:33]: Remove comment since you abstracted

File: lib/presentation/auth/auth_screen.dart [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+          state.maybeMap(
+              smsCodeSent: (_) => _toPage(1),
+              loggedIn: (loggedInState) {
+                if (loggedInState.isNewUser) {
+                  // TODO - Handle if is existing user
+                } else {
+                  _toPage(2);
+                }
+              },
+              verificationCompleted: (_) {},
```
Comment (Xazin) [inline lib/presentation/auth/auth_screen.dart:80]: Why is this empty?

File: lib/presentation/auth/auth_screen.dart [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+              },
+              photoUpdateDone: (_) {
+                _authDone(context);
+              },
+              orElse: () {});
+        },
+        child: Scaffold(
+          resizeToAvoidBottomInset: true,
+          appBar: _currentPage == 0
+              ? CustomAppBar(
```
Comment (Xazin) [inline lib/presentation/auth/auth_screen.dart:104]: We don't have CustomAppBar anymore, need to rebase on the remote development branch and use CleanAppBar.

File: lib/infrastructure/auth/firebase_auth_repository.dart (hunk 1 above)
Comment (wizlif) [reply lib/infrastructure/auth/firebase_auth_repository.dart:91]: Let me do that in this P.R

File: lib/presentation/auth/auth_screen.dart (hunk 6 above)
Comment (wizlif) [reply lib/presentation/auth/auth_screen.dart]: Removed it, it was being handled in the Pin input page

**Comment by Xazin** [comment]
Closed for now! See: #47

**Pull request closed without merging**